lib/Models/SenapsLocationsCatalogItem.ts

```diff
@@ -0,0 +1,279 @@
+    <br/>
+    <chart
+      id='{{id}}'
+      title='{{id}}'
+      sources='https://senaps.io/api/sensor/v2/observations?streamid={{#terria.urlEncodeComponent}}{{streamIds}}{{/terria.urlEncodeComponent}}&limit=1440&media=csv&csvheader=false&sort=descending,https://senaps.io/api/sensor/v2/observations?streamid={{#terria.urlEncodeComponent}}{{streamIds}}{{/terria.urlEncodeComponent}}&limit=7200&media=csv&csvheader=false&sort=descending'
+      source-names='1d,5d'
+      downloads='https://senaps.io/api/sensor/v2/observations?streamid={{#terria.urlEncodeComponent}}{{streamIds}}{{/terria.urlEncodeComponent}}&limit=1440&media=csv&csvheader=false&sort=descending,https://senaps.io/api/sensor/v2/observations?streamid={{#terria.urlEncodeComponent}}{{streamIds}}{{/terria.urlEncodeComponent}}&limit=7200&media=csv&csvheader=false&sort=descending'
+      download-names='1d,5d'
+    >
+    </chart>
+  {{/hasStreams}}
+  {{^hasStreams}}
+    No streams ar this locations
+    <br/><br/>
+  {{/hasStreams}}
+  `
+      });
+      senapsLocationsCatalogItem.setTrait(
+        "definition",
+        "featureInfoTemplate",
+        featureInfo
+      );
+
+      gjci.loadMapItems();
+      return new SenapsLocationsStratum(senapsLocationsCatalogItem, gjci);
+    } catch (e) {
+      const msg =
+        e.statusCode === 401
+          ? "models.senaps.missingKeyErrorMessage"
+          : "models.senaps.generalErrorMessage";
+      throw new TerriaError({
+        title: i18next.t("models.senaps.retrieveErrorTitle"),
+        message: i18next.t(msg)
+      });
+    }
+  }
+
+  get dataSource(): GeoJsonCatalogItem {
+    return this.geojsonItem;
+  }
+}
+
+StratumOrder.addLoadStratum(SenapsLocationsStratum.stratumName);
+
+class SenapsLocationsCatalogItem extends AsyncMappableMixin(
+  CatalogMemberMixin(CreateModel(SenapsLocationsCatalogItemTraits))
+) {
+  static readonly type = "senaps-locations";
+
+  readonly baseUrl = "https://senaps.io/api/sensor/v2";
+
+  get type() {
+    return SenapsLocationsCatalogItem.type;
+  }
+
+  get typeName() {
+    return i18next.t("models.senaps.name");
+  }
+
+  readonly canZoomTo = true;
+
+  protected forceLoadMapItems(): Promise<void> {
+    return SenapsLocationsStratum.load(this).then(stratum => {
+      if (stratum === undefined) return;
+      runInAction(() => {
+        this.strata.set(SenapsLocationsStratum.stratumName, stratum);
+      });
+    });
+  }
+
+  @computed get geoJsonItem() {
+    const stratum = <SenapsLocationsStratum>(
+      this.strata.get(SenapsLocationsStratum.stratumName)
+    );
+    return isDefined(stratum) ? stratum.dataSource : undefined;
+  }
+
+  @computed get mapItems() {
+    if (isDefined(this.geoJsonItem)) {
+      return this.geoJsonItem.mapItems.map(mapItem => {
+        mapItem.show = this.show;
+        return mapItem;
+      });
+    }
+    return [];
+  }
+
+  protected forceLoadMetadata(): Promise<void> {
+    return Promise.resolve();
+  }
+
+  _constructLocationsUrl() {
```

Comment:
Yeah one of those - i got confused making my tests pass and having them callable from within my test, I'l relook at it

Reply:
So if I set to `protected` or `private` I can't access the methods in the  `SenapsLocationsStratum` class where I currently am . 

Because they're currently being used in the `static` `load` method I don't yet access to the full  `SenapsLocationsStratum` as `this` or whatver.

So I'm inclined to leave it as is.

Reply:
Sounds fine to me!